### PR TITLE
34905/one to one mapping rules

### PR DESCRIPTION
**Pull request by calmacx**
alternative to #114

### CODE REVIEW

File: api/mapping/services.py [hunk 1]
```diff
@@ -17,67 +16,8 @@
...
-
-    if concept_relation.concept_id_2 != concept_relation.concept_id_1:
-        concept = Concept.objects.get(
-            concept_id=concept_relation.concept_id_2
-        )
-        return concept
-    else:
-        #may need some warning if this ever happens?
-        return source_concept
-
```
Comment (calmacx) [inline api/mapping/services.py:80]: deleted as moved to services_rules.py

File: api/mapping/views.py [hunk 2]
```diff
@@ -429,7 +445,10 @@ def generate(self, request):
                 values = field.scanreportvalue_set.all().filter(conceptID__gte=0)
 
                 # map the source value to the raw value
-                concepts = {value.value: value.conceptID for value in values}
+                #concepts = {value.value: value.conceptID for value in values}
+                #messages.warning(request, concepts)
+                #print (concepts)
+
```
Comment (calmacx) [inline api/mapping/views.py:451]: this code is all redundant now anyway

File: api/mapping/views.py [hunk 3]
```diff
@@ -578,21 +598,25 @@ def download_structural_mapping(self, request, return_type="json"):
...
-            if rule.term_mapping:
-                output["term_mapping"] = json.loads(rule.term_mapping)
-
+            #if we are to do term_mapping for this rule
+            
+            key = concept.content_object.value
+            #if it's flagged to use the source_concept, use this
+            value = concept.concept.concept_id
+            term_mapping = {key:value}
+
```
Comment (calmacx) [inline api/mapping/views.py:615]: untested, download json wont be working with this PR

File: api/mapping/views.py [hunk 4]
```diff
@@ -1470,23 +1497,53 @@ def save_scan_report_value_concept(request):
...
+                    messages.error(request,
+                                   "You could try {} ({}) ?".format(concept.concept_id,
+                                   concept.concept_name))
+                    
+                    return redirect("/values/?search={}".format(scan_report_value.scan_report_field.id))
+            else:
+                #otherwise, if this is a standard concept (source_concept.standard_concept=='S')
+                #we are good and set concept == source_concept
+                concept = source_concept
+
```
Comment (calmacx) [inline api/mapping/views.py:1531]: some validation that might be better suited elsewhere

File: api/data/models/concept.py [hunk 5]
```diff
@@ -49,3 +49,4 @@ class Concept(models.Model):
     class Meta:
         managed = False
         db_table = 'omop"."concept'
+        #db_table = 'concept'
```
Comment (spco) [inline api/data/models/concept.py:52]: Please remove this commented code, and below.

File: api/mapping/templates/mapping/scanreportvalue_list.html [hunk 6]
```diff
@@ -25,7 +25,11 @@ <h2>Values</h2>
                 <th scope="col">Value</th>
                 <th scope="col">Value Description</th>
                 <th scope="col">Frequency</th>
-                <th scope="col">Concepts</th>
+                <th scope="col">Concepts
+		  <!--- 
+		      (<span class="text-success"> Standard </span> / <span class='text-warning'> Non-Standard </span>)
+		    --->
+		</th>
```
[the remark this answers is not in the document]
Comment (calmacx) [reply api/mapping/templates/mapping/scanreportvalue_list.html:32]: this can go

File: api/mapping/services.py [hunk 7]
```diff
@@ -17,79 +16,6 @@
     DataDictionary
 )
 
-from coconnect.tools.omop_db_inspect import OMOPDetails
-
-
-
-def get_concept_from_concept_code(concept_code,
```
Comment (calmacx) [inline api/mapping/services.py:24]: make sure doesnt mess up NLP running

File: api/mapping/services.py (hunk 1 above)
Comment (calmacx) [reply api/mapping/services.py:93]: make into separate PR

File: api/mapping/services_nlp.py [hunk 8]
```diff
@@ -10,7 +10,7 @@
 
 from .models import (ScanReport, ScanReportAssertion, ScanReportConcept,
                      ScanReportField, ScanReportValue)
-from .services import find_standard_concept, get_concept_from_concept_code
+from .services_rules import find_standard_concept, get_concept_from_concept_code
```
Comment (calmacx) [inline api/mapping/services_nlp.py:13]: make into separate PR

File: api/mapping/templates/mapping/mappingrulesscanreport_list.html [hunk 9]
```diff
@@ -68,8 +72,8 @@ <h6>Filter On Source Table</h6>
 	<th scope="col">Source Table</th>
 	<th scope="col">Source Field</th>
 	<th scope="col">Term Map</th>
-	<th scope="col">Operation</th>
-	<th scope="col">Approved</th>
```
Comment (calmacx) [inline api/mapping/templates/mapping/mappingrulesscanreport_list.html:72]: fix this table in a separate PR
